hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java

```diff
@@ -278,4 +278,35 @@ public void testCreateSnapshotParameterMissing() throws Exception {
       shell.close();
     }
   }
+
+  @Test
+  public void testDeleteBucketWithSnapshot() throws Exception {
+    String volume = "vol-" + RandomStringUtils.randomNumeric(5);
+    String bucket = "buc-" + RandomStringUtils.randomNumeric(5);
+    String key = "key-" + RandomStringUtils.randomNumeric(5);
+    String snapshotName = "snap-" + RandomStringUtils.randomNumeric(5);
+
+    String testVolBucket = OM_KEY_PREFIX + volume + OM_KEY_PREFIX + bucket;
+    String testKey = testVolBucket + OM_KEY_PREFIX + key;
+
+    createVolBuckKey(testVolBucket, testKey);
+
+    OzoneFsShell shell = new OzoneFsShell(clientConf);
+    try {
+      int res = ToolRunner.run(shell,
+              new String[]{"-createSnapshot", testVolBucket, snapshotName});
+      // Asserts that create request succeeded
+      assertEquals(0, res);
+
+      res = ToolRunner.run(shell,
+              new String[]{"-rm", "-r", "-skipTrash", testKey});
+      assertEquals(0, res);
+
+      res = ToolRunner.run(shell,
+              new String[]{"-rm", "-r", "-skipTrash", testVolBucket});
+      assertEquals(1, res);
+    } finally {
```

Comment:
Should there be one more check (or two) to verify that the bucket and the snapshot still exist?

Reply:
I have added checks for bucket and snapshot. The snapshot check is failing presently due to - https://issues.apache.org/jira/browse/HDDS-7805